src/passes/OptimizeInstructions.cpp

```diff
@@ -1733,6 +1733,39 @@ struct OptimizeInstructions
         !effects(curr->right).hasSideEffects()) {
       return curr->left;
     }
+    {
+      // C1 - (x + C2)  ==>  (C1 - C2) - x
+      Const *c1, *c2;
+      Expression* x;
+      if (matches(curr,
+                  binary(Sub, ival(&c1), binary(Add, any(&x), ival(&c2))))) {
+        left->value = c1->value.sub(c2->value);
+        curr->right = x;
+        return curr;
+      }
+      // C1 - (x - C2)  ==>  (C1 + C2) - x
+      if (matches(curr,
+                  binary(Sub, ival(&c1), binary(Sub, any(&x), ival(&c2))))) {
+        left->value = c1->value.add(c2->value);
+        curr->right = x;
+        return curr;
+      }
+      // C1 - (C2 - x)  ==>   x + (C1 - C2)
+      if (matches(curr,
+                  binary(Sub, ival(&c1), binary(Sub, ival(&c2), any(&x))))) {
+        left->value = c1->value.sub(c2->value);
+        if (left->value.isNegative()) {
+          // -C1 - (C2 - x)  ==>  x - (C1 - C2)`
```

Comment:
Oops, looks my typo made it in 😅 
```suggestion
          // -C1 - (C2 - x)  ==>  x - (C1 - C2)
```